gemonlinedb/src/common/AMC13Configuration.cc

```diff
@@ -134,5 +134,34 @@ namespace gem {
             return dom;
         }
 
+        template<>
+        AMC13Configuration DBInterface::convertRow<AMC13Configuration>(
+            const xdata::Table &table, xdata::Table::Row &row)
+        {
+            AMC13Configuration config;
+
+            auto value = row.getField("FED_ID");
```

Comment:
This method should be refactored: construct a list of fields and iterate over a list calling 
config.SetValue("FieldName", value). A corresponding single method in config has to be implemented instead of the three used here. 
An exception has to be thrown in case value is a `nullptr` or empty.

Reply:
A list of fields could only iterate over two of the values since the third is a `string` (which is why this specialization is needed in the first place). I don't think it's worth the additional complexity.

Reply:
hmmm... 
```c++
try {
   int val_i = std::stoi(value);
    config.setValue("FieldName", val_i);
} catch (const std::invalid_argument& e) {
    config.setValue("FieldName", value);
}
```

Reply:
* Doesn't handle the case where `value` is a valid number but needs to be set as a string
* Call `setValue(std::string, std::string)` for integer fields when the integer isn't valid and it should fail
* The functions `config.setValue(std::string, X)` would basically contain a switch statement delegating to the individual functions already called in the current implementation, which would add one layer for very little gain

Reply:
Somewhat cleaner with a macro: https://github.com/cms-gem-daq-project/cmsgemos/pull/237/files#diff-5e87771af216d4fce8dfa4eb565b7070R144

Reply:
Fixed in 05feeefcc80c0dd45e0038a82c8c97e45fabc0e9 (#237)